src/main/java/org/openlmis/stockmanagement/service/StockCardAggregate.java

```diff
@@ -92,14 +92,19 @@ public Map<String, Integer> getAmounts(LocalDate startDate, LocalDate endDate) {
 
     return isEmpty(filteredLineItems) ? new HashMap<>() : filteredLineItems.stream()
         .map(lineItem -> {
```

Comment:
Consider separating this whole 'map' lambda into couple of methods.
The totalTagsValues could be extracted to calculateTagValuesForStockAdjustments.
That other calculation which involves value:int, tags:List and stream at 104, could be extracted to calculateTagValuesForLineItem.

At the end the final 'map' lambda could be something like:
`
List<ImmutablePair<String,Integer>> totalTagsValues = new ArrayList<>(calculateTagValuesForStockAdjustments(...));
totalTagsValues.addAll(calculateTagValuesForLineItem(...));
return totalTagsValues;
`

Reply:
I have created two methods to separate whole lambda, and upgraded final lambda as mentioned in comment.